core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala

```diff
@@ -54,6 +54,11 @@ class LogCleanerManagerTest extends Logging {
     override def allCleanerCheckpoints: Map[TopicPartition, Long] = {
       cleanerCheckpoints.toMap
     }
+
+    override def updateCheckpoints(dataDir: File, update: Option[(TopicPartition,Long)]): Unit = {
+      val (tp, offset) = update.get
```

Comment:
LogCleanerManager.updateCheckpoints() can be called with `update=None`. In this case, the method removes topic partitions that no longer exist from the checkpoint file. It looks like the test suite does not deal with test cases that remove partitions when testing with mock log cleaner manager. However, it's useful to handle this case explicitly here so that it is easier to debug if code changes, etc.

I think it's enough to throw an exception here if update=None with the explicit message that this case is not handled (yet) by LogCleanerManagerMock (you can do this by calling `update.getOrElse(throw new...)). Or maybe you have other ideas.